framework/src/engine/legacy/storage.ts

```diff
@@ -12,51 +12,147 @@
+			throw error;
+		}
 	}
 
-	// eslint-disable-next-line @typescript-eslint/require-await
-	public async getBlockByHeight(_height: number): Promise<Buffer> {
-		return Buffer.alloc(0);
+	public async getBlockByHeight(_height: number): Promise<BlockJson> {
+		try {
+			const id = await this._db.get(Buffer.concat([DB_KEY_BLOCK_HEIGHT, intToBuffer(_height, 4)]));
+			return this.getBlockByID(id);
+		} catch (error) {
+			if (error instanceof NotFoundError) {
+				throw new NotFoundError(`Specified height ${_height} does not exist.`);
+			}
+
+			this._logger.error({ err: error as Error }, 'Failed to get block by height.');
+			throw error;
+		}
 	}
 
-	// eslint-disable-next-line @typescript-eslint/require-await
-	public async getBlocksByHeightBetween(_fromHeight: number, _toHeight: number): Promise<Buffer[]> {
-		return [];
+	public async getBlocksByHeightBetween(
+		_fromHeight: number,
+		_toHeight: number,
+	): Promise<BlockJson[]> {
+		try {
+			const blockIDs = await this._getBlockHeadersByHeightBetween(_fromHeight, _toHeight);
+			const blocks: BlockJson[] = [];
+
+			for (const blockID of blockIDs) {
+				const id = blockID;
+				const block = await this.getBlockByID(id);
+				blocks.push(block);
+			}
+
+			return blocks;
+		} catch (error) {
+			this._logger.error({ err: error as Error }, 'Failed to get blocks by height between.');
+			throw error;
+		}
 	}
 
-	// eslint-disable-next-line @typescript-eslint/require-await
 	public async isBlockPersisted(_blockID: Buffer): Promise<boolean> {
-		return true;
+		try {
+			return await this._db.has(Buffer.concat([DB_KEY_BLOCK_ID, _blockID]));
+		} catch (error) {
+			if (error instanceof NotFoundError) {
+				throw new NotFoundError(`Specified key ${_blockID.toString('hex')} does not exist.`);
+			}
+
+			this._logger.error({ err: error as Error }, 'Failed to check block persistence.');
+			throw error;
+		}
 	}
 
-	// eslint-disable-next-line @typescript-eslint/require-await
 	public async isBlockHeightPersisted(_height: number): Promise<boolean> {
-		return true;
-	}
-
-	/*
-		Save Block
-	*/
-	public async saveBlock(
-		_id: Buffer,
-		_height: number,
-		_block: Buffer,
-		// eslint-disable-next-line @typescript-eslint/no-empty-function
-	): Promise<void> {}
+		try {
+			return await this._db.has(Buffer.concat([DB_KEY_BLOCK_HEIGHT, intToBuffer(_height, 4)]));
+		} catch (error) {
+			if (error instanceof NotFoundError) {
+				throw new NotFoundError(`Specified height ${_height} does not exist.`);
+			}
+
+			this._logger.error({ err: error as Error }, 'Failed to check block height persistence.');
+			throw error;
+		}
+	}
+
+	public async saveBlock(_id: Buffer, _height: number, _block: Buffer): Promise<void> {
+		try {
+			const batch = new Batch();
+			batch.set(Buffer.concat([DB_KEY_BLOCK_ID, _id]), _block);
+			batch.set(Buffer.concat([DB_KEY_BLOCK_HEIGHT, intToBuffer(_height, 4)]), _id);
+
+			await this._db.write(batch);
+		} catch (error) {
```

Comment:
In general, I think no need to catch and log error because it will be log on upper later

Reply:
Ahh ok. Will update all methods.